inference-engine/include/ie_blob.h

```diff
@@ -199,6 +200,17 @@ class INFERENCE_ENGINE_API_CLASS(Blob) {
      */
     virtual LockedMemory<const void> cbuffer() const noexcept = 0;
 
+    /**
+     * @brief Creates a blob describing given ROI object based on the current blob with memory sharing.
+     *
+     * Note: default implementation throws "not implemented" exception.
+     *
+     * @param roi A ROI object inside of the current blob.
+     *
+     * @return A shared pointer to the newly created ROI blob.
+     */
+    virtual Blob::Ptr createROI(const ROI& roi) const;
```

Comment:
why we cannot make this method pure virtual?

Reply:
To avoid breaking existing code (remote blobs in some plugins). Default implementation just throws not implemented exception, which, in my opinion is OK (that's I would do for such cases).